src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -84,11 +91,23 @@ impl SstableStreamIterator {
         // `next_block()` loads a new block (i.e., `block_iter` is not `None`), then `block_iter` is
         // also valid and pointing on the block's first KV-pair.
 
-        if let (Some(block_iter), Some(seek_key)) = (self.block_iter.as_mut(), seek_key) {
-            block_iter.seek(seek_key);
+        while let Some(block_iter) = self.block_iter.as_mut() {
+            if let Some(seek_key) = seek_key {
+                block_iter.seek(seek_key);
```

Comment:
> I think we don't need to do `block_iter.seek` in a while loop because keys in all blocks except for the first block are guaranteed to be larger than the seek_key.

why?